Sources/MapboxCoreNavigation/RouteController.swift

```diff
@@ -256,12 +256,17 @@ open class RouteController: NSObject {
         }
 
         let routeRequest = Directions().url(forCalculating: progress.routeOptions).absoluteString
-        let routes = Routes(routesResponse: routeJSONString,
-                            routeIndex: 0,
-                            legIndex: UInt32(progress.legIndex),
-                            routesRequest: routeRequest)
+        // Based on MBNNRouteIndex.routeId documentation.
+        // FIXME: Distinguish onboard routes with “local@” prefix.
+        let routeResponseIdentifier = indexedRouteResponse.routeResponse.identifier ?? ""
+        let routeIndices = (indexedRouteResponse.routeResponse.routes ?? []).enumerated().map { index, route in
+            RouteIndex(routeId: "\(routeResponseIdentifier)#\(index)", indexInResponse: UInt32(index))
+        }
```

Comment:
This route identifier generation code is problematic. The format is hardcoded here based on a documentation comment buried deep in MapboxNavigationNative. There doesn’t seem to be a way to distinguish a remote route from a local route. That was previously by design. We probably need MapboxDirections to persist unrecognized properties in order to have access to the route identifier that the navigator expects: mapbox/mapbox-directions-swift#637. Or is there some other way for this method to determine the source of the route?

/cc @Udumft @SiarheiFedartsou

Reply:
We introduced Routes.parse API on our side - we will encapsulate this logic there. I tagged you in that PR.

Reply:
cc @DenisPryt